soap/src/java/com/zimbra/soap/mail/type/ContactFrequencyData.java

```diff
@@ -14,19 +13,19 @@
 
     public ContactFrequencyData() {}
 
-    public ContactFrequencyData(String frequencyBy, List<ContactFrequencyDataPoint> dataPoints) {
-        setFrequencyBy(frequencyBy);
+    public ContactFrequencyData(ContactFrequencyGraphSpec graphSpec, List<ContactFrequencyDataPoint> dataPoints) {
+        setGraphSpec(graphSpec);
         setDataPoints(dataPoints);
     }
 
-    @XmlAttribute(name=MailConstants.A_CONTACT_FREQUENCY_BY, required=true)
-    private String frequencyBy;
+    @XmlElement(name=MailConstants.A_CONTACT_FREQUENCY_GRAPH_SPEC, required=true)
```

Comment:
Codacity likes to have fields before constructors etc.  Worth following that convention just to shut Codacity up?  Or if we don't think it is worth the hassle, perhaps we should change the Codacity rules.